### DISCUSSION TITLE
Feature Request: Set up image transforms from a config file

### DISCUSSION THREAD

**Discussion by romainpoirier** [craftcms/cms#8790] [Ideas]
### Description

It would be nice to be able to set up assets transforms from a config file, like we can do for routing. Set up these transforms from the CP is not the most efficient way, and is dependent of the database.

I would like to take benefits of the new automatic generation of transforms feature of [Imager X plugin](https://imager-x.spacecat.ninja/usage/generate.html), but for a large project with tons of different transforms (including variants for breakpoints), it costs a lot of time. So much that currently I'm setting up my transforms directly from the Twig templates.

**Comment by piotrpog** [comment]
Actually, you can do this already.

First, set array with transform settings in `general.php`. Then you can access this array in twig using 

```
{{craft.app.config.general.variable_name}}
```

**Comment by romainpoirier** [comment]
Yep, thank you. I was thinking about a way like this. But as this is not an official way, I think this can't work with Imager X, as it would not be registered as known transforms.
